Description: x32 and hppa require -fPIC
 x32 detection is currently Debian-specific.
Author: Stefano Rivera <stefanor@debian.org>
Bug-Debian: https://bugs.debian.org/760267
Forwarded: no

--- a/rpython/translator/platform/__init__.py
+++ b/rpython/translator/platform/__init__.py
@@ -270,7 +270,10 @@
     # Only required on armhf and mips{,el}, not armel. But there's no way to
     # detect armhf without shelling out
     if (platform.architecture()[0] == '64bit'
-            or platform.machine().startswith(('arm', 'mips', 'ppc'))):
+            or platform.machine().startswith(
+                ('arm', 'mips', 'parisc', 'ppc'))
+            or getattr(getattr(sys, 'implementation', sys), '_multiarch',
+                       None) == 'x86_64-linux-gnux32'):
         host_factory = LinuxPIC
     else:
         host_factory = Linux
